pyrogram/client/methods/chats/join_chat.py

```diff
@@ -36,7 +37,7 @@ def join_chat(self,
         match = self.INVITE_LINK_RE.match(chat_id)
 
         if match:
-            return self.send(
+            chat = self.send(
```

Comment:
Here we could just use `Chat._parse_chat_chat` or `Chat._parse_channel_chat depending on the result...